server/channels/api4/ip_filtering.go

```diff
@@ -81,7 +81,9 @@ func applyIPFilters(c *Context, w http.ResponseWriter, r *http.Request) {
 
 	auditRec.Success()
 
-	go c.App.SendIPFiltersChangedEmail(c.AppContext, c.AppContext.Session().UserId)
+	if err := go c.App.SendIPFiltersChangedEmail(c.AppContext, c.AppContext.Session().UserId); err != nil {
```

Comment:
That is not valid go code :wink: 

You need to use an anonymous function. Something like

``golang
go func() {
...
}
```

Reply:
I have changed it to
```go
go func() {
		if err := c.App.SendIPFiltersChangedEmail(c.AppContext, c.AppContext.Session().UserId); err != nil {
			c.Err = model.NewAppError("SendIPFiltersChangedEmail", "api.context.ip_filtering.send_email.app_error", nil, "", http.StatusInternalServerError).Wrap(err)
		}
	}()
```

However I seem to be getting a slicing error 
```bash
ERRO Running error: can't run linter goanalysis_metalinter
goanalysis_metalinter: buildir: package "slices" (isInitialPkg: false, needAnalyzeSource: true): Cannot range over: func(yield func(E) bool)
```

When changing the line
```go
w.Write(json)
```

To

```go
if _, err := w.Write(json); err != nil {
		c.Logger.Warn("Error while writing response", mlog.Err(err))
	}
```

Any idea why that may be the case?

Reply:
Similar answer as in https://github.com/mattermost/mattermost/issues/28399#issuecomment-2395006772: Are you using go1.23?

Reply:
That seems like it was the issue... 

I added the import for mlog and changed to a go func